sycl/test/regression/unnamed-lambda.cpp

```diff
@@ -0,0 +1,10 @@
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-device-only -c %s -o %t.temp
+
```

Comment:
Can you add a comment here as well?  What is being tested here? That unnamed lambdas still work? Do we not have existing tests for this?

Reply:
We're testing the reproducer that this fixes.  This only got caught in post-commit via a huge example.